### PR TITLE
Sale producer to BI

### PR DESCRIPTION
**Pull request by Jack-Is-2048**

### CODE REVIEW

File: src/main/java/com/mercuryCyclists/Sales/KafkaConfig/Producer.java [hunk 1]
```diff
@@ -11,21 +12,22 @@
...
-     @Bean
-     public KafkaTemplate<String, String> kafkaTemplate(ProducerFactory<String, String> produceFactory) {
-         return new KafkaTemplate<>(produceFactory);
-     }
+    @Bean
+    public KafkaTemplate<String, String> kafkaTemplate(ProducerFactory<String, String> produceFactory) {
+        return new KafkaTemplate<>(produceFactory);
+    }
 }
+*/
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/KafkaConfig/Producer.java]: rm commented stuff 😠

Comment (Jack-Is-2048) [reply src/main/java/com/mercuryCyclists/Sales/KafkaConfig/Producer.java]: No 😠

File: src/main/java/com/mercuryCyclists/Sales/entity/SaleEvent.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package com.mercuryCyclists.Sales.entity;
+
+public class SaleEvent {
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/entity/SaleEvent.java:6]: use an all args constructor annotation?

Comment (Jack-Is-2048) [reply src/main/java/com/mercuryCyclists/Sales/entity/SaleEvent.java:6]: I mean we can but currently uses default + setters

File: src/main/java/com/mercuryCyclists/Sales/service/OnlineSaleService.java [hunk 3]
```diff
@@ -119,7 +145,18 @@ public OnlineSale registerBackorder(OnlineSale onlineSale){
...
+
+        // Create and populate sale event
+        SaleEvent saleEvent = new SaleEvent();
+        saleEvent.setProductName(product.get("name").getAsString());
+        saleEvent.setQuantity(onlineSale.getQuantity());
+        Double pricePerProduct = product.get("price").getAsDouble();
+        Double totalSalePrice = pricePerProduct * onlineSale.getQuantity();
+        saleEvent.setPrice(totalSalePrice);
+
+        System.out.println(saleEvent);
```
Comment (pedrooste) [inline src/main/java/com/mercuryCyclists/Sales/service/OnlineSaleService.java]: can we make this a generic method in sale service, then call it from all.
rm println also

**Review by pedrooste — CHANGES_REQUESTED** [review]
Looks good! Small changes

**Comment by pedrooste** [comment]
also doco on new event class

**Review by pedrooste — APPROVED** [review]
Looks good

**Pull request merged**